test-suite/Math/NumberTheory/Recurrencies/PentagonalTests.hs

```diff
@@ -48,10 +53,30 @@ partitionProperty1 (Positive n) =
                      takeWhile (\m -> n - m >= 0) .
                      tail $ pents)
 
+-- | Check that
+-- @partition :: Math.NumberTheory.Moduli.Mod n == map (`mod` n) partition@.
+partitionProperty2 :: NonNegative Integer -> Nat -> Bool
+partitionProperty2 (NonNegative m) n = driver
+  where
+    m' = fromIntegral m
+    nat = natVal @n
+    p :: KnownNat p => p
+    p = case someNatVal n of
+        SomeNat (_ :: Proxy t) -> t
+    driver :: KnownNat p => Bool
+    driver = take m' (partition :: KnownNat n => [Mod p]) ==

```

Comment:
@Bodigrim I haven't made this work yet, but the idea is present and this and the next lines. I'm not sure how I'll reconcile the `[SomeMod]` produced by helper with the `[Mod n]` that `partition` return.

Reply:
We can either apply `fromInteger :: Integer -> Mod p` to `partition :: [Integer]` and compare with `partition :: [Mod p]`, or on contrary extract integers from `partition :: [Mod p]` by `getVal` and compare with `map (flip mod p) partition :: [Integer]`. 

Or wrap everything up to `SomeMod`, employing its `Eq` instance, applying `SomeMod` constructor to `Mod p` and `flip modulo p` to `Integer`.

Reply:
>or on contrary extract integers from partition :: [Mod p] by getVal and compare with map (flip mod p) partition :: [Integer]

I'll need to somehow reflect the `Natural` passed to the test as an argument into a `Nat`. `someNatVal` does `:: Natural -> SomeNat`, but I'm not sure how to do `SomeNat -> Nat`, but at the type level.

Reply:
Actually you cannot pass any term of type `Nat` to `partitionProperty2`, because `Nat` is an uninhabited type (but "inhabited" kind). So we should go other way around: pass `Natural` to `partitionProperty` and reflect it on type level as `Nat`, e. g., following pattern from [modulo](https://github.com/cartazio/arithmoi/blob/master/Math/NumberTheory/Moduli/Class.hs#L216).

Reply:
I did it just before you posted this, now I'm fixing the fact that the test will fail because `smallcheck:                     spec: internal error: Unable to commit 1048576 bytes of memory`. This is related to below's comments about using `Map Int a` in `partition`. I'll commit the corrected test for now.